analyzer/analyzer.spicy

```diff
@@ -19,6 +22,23 @@ public type Message = unit {
     sname: bytes &size=64 &convert=$$.split1(b"\0")[0].decode();
     file_n: bytes &size=128 &convert=$$.split1(b"\0")[0].decode();
     options: Options;
+
+    # Correct encoding for chaddr
+    var chaddr_iter: iterator<bytes>;  # Iterator on chaddr
+    var chaddr_str: string = "";  # Accumulator string for chaddr correct encoding
+    var i: uint8 = 0;
+    on chaddr {
+        self.chaddr_iter = begin($$);
+        # Iterate up to hlen, the hardware address length given in the message
+        while ( self.i < self.hlen ) {
+            if ( self.i > 0 ) {
+                self.chaddr_str = self.chaddr_str + ":";
+            }
+            self.chaddr_str = self.chaddr_str + ("%02x" % *(self.chaddr_iter++));
+            ++self.i;
+        }
+    }
+    
```

Comment:
This introduces `chaddr_iter` and `i` as unit variables which are not that interesting to users. Let's instead make them `local` in the `on chaddr` hook.

```suggestion

    var chaddr_str: string;
    on chaddr {
        # TODO(bbannier): Perform this conversion on the fly as part of parsing
        # `chaddr` once https://github.com/zeek/spicy/issues/1261 is fixed.

        # Iterate up to hlen, the hardware address length given in the message
        local i = 0;
        while ( i < self.hlen ) {
            if ( i > 0 ) {
                self.chaddr_str = self.chaddr_str + ":";
            }
            self.chaddr_str = self.chaddr_str + ("%02x" % *self.chaddr.at(i));
            ++i;
        }
    }

```